src/routes/safe/components/AddressBook/index.jsx

```diff
@@ -25,36 +25,76 @@ import {
   AB_ADDRESS_ID,
   ADDRESS_BOOK_ROW_ID,
   EDIT_ENTRY_BUTTON,
-  generateColumns, REMOVE_ENTRY_BUTTON, SEND_ENTRY_BUTTON,
+  generateColumns,
+  REMOVE_ENTRY_BUTTON,
+  SEND_ENTRY_BUTTON,
 } from '~/routes/safe/components/AddressBook/columns'
-import loadAddressBook from '~/logic/addressBook/store/actions/loadAddressBook'
-import { getAddressBookListSelector } from '~/logic/addressBook/store/selectors'
 import Col from '~/components/layout/Col'
 import ButtonLink from '~/components/layout/ButtonLink'
-
+import CreateEditEntryModal from '~/routes/safe/components/AddressBook/CreateEditEntryModal'
+import { getAddressBook } from '~/logic/addressBook/store/selectors'
+import type { AddressBookEntryType } from '~/logic/addressBook/model/addressBook'
+import DeleteEntryModal from '~/routes/safe/components/AddressBook/DeleteEntryModal'
+import { updateAddressBookEntry } from '~/logic/addressBook/store/actions/updateAddressBookEntry'
+import { removeAddressBookEntry } from '~/logic/addressBook/store/actions/removeAddressBookEntry'
+import { addAddressBookEntry } from '~/logic/addressBook/store/actions/addAddressBookEntry'
+import SendModal from '~/routes/safe/components/Balances/SendModal'
+import { safeSelector } from '~/routes/safe/store/selectors'
+import { extendedSafeTokensSelector } from '~/routes/safe/container/selector'
 
 type Props = {
   classes: Object
 }
 
-
-const AddressBookTable = ({
-  classes,
-}: Props) => {
+const AddressBookTable = ({ classes }: Props) => {
   const columns = generateColumns()
   const autoColumns = columns.filter((c) => !c.custom)
   const dispatch = useDispatch()
-  useEffect(() => {
-    dispatch(loadAddressBook())
-  }, [])
 
-  const addressBook = useSelector(getAddressBookListSelector)
+  const addressBookObj = useSelector(getAddressBook)
+  const addressBook = List(addressBookObj)
```

Comment:
can we maybe write a selector that would return a list? we do the same for the safe list

Reply:
Wasn't able to do that, seems that the selectors makes a cache of the initial status of the map and then it nevers updates again

Reply:
there is `safesListSelector` in '~/routes/safe/store/selectors' which does just that. Could you maybe try again and message me on slack if you face any problem?

Reply:
done